siege_utilities/geo/s2_utils.py

```
@@ -0,0 +1,676 @@
+        return _s2.CellId.from_token(value)
+    # Try integer-like (covers Python int, numpy.int64, numpy.uint64).
+    try:
+        return _s2.CellId(int(value))
+    except (TypeError, ValueError):
+        raise TypeError(
+            f"Expected CellId, int64, or token string, got "
+            f"{type(value).__name__}"
+        )
+
+
+def _coerce_polygon(geometry):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_coerce_polygon`

(ANN202)